### PR TITLE
Sponge port

### PR DESCRIPTION
**Pull request by SamB440**
Adds support for Sponge, project is now multi-module

### CODE REVIEW

File: paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import java.io.IOException
+import java.net.MalformedURLException
+import java.util.logging.FileHandler
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class LemonadeStandPaper: JavaPlugin(), LemonadeStand<OfflinePlayer>, Listener {
+
+    lateinit var transactionLogger: Logger
+        private set
```
Comment (StealWonders) [inline paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt:31]: Why this weird formatting?

File: sponge/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    apiVersion("11.0.0-SNAPSHOT")
+    loader {
+        name(PluginLoaders.JAVA_PLAIN)
+        version("1.0.0-SNAPSHOT")
+    }
+    plugin("lemonadestand") {
+        displayName("LemonadeStand")
+        entrypoint("org.empirewar.lemonadestand.sponge.LemonadeStandSponge")
+        description("Accept incoming Ko-Fi webhooks and allow further processing.")
+        license("GPL-3.0")
```
Comment (StealWonders) [inline sponge/build.gradle.kts:24]: Should be MIT

File: sponge/src/main/kotlin/org/empirewar/lemonadestand/sponge/LemonadeStandSponge.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+class LemonadeStandSponge @Inject constructor(
+    pluginContainer: PluginContainer,
+    @ConfigDir(sharedRoot = false)
+    private val dataFolder: Path,
+    private val logger: Logger
+) : LemonadeStand<User> {
+
+    lateinit var transactionLogger: Logger
+        private set
```
Comment (StealWonders) [inline sponge/src/main/kotlin/org/empirewar/lemonadestand/sponge/LemonadeStandSponge.kt:44]: Weird formatting?

File: build-logic/src/main/kotlin/buildlogic.java-common-conventions.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        trimTrailingWhitespace()
+    }
+}
+
+repositories {
+    // Use Maven Central for resolving dependencies.
+    mavenCentral()
+    maven("https://repo.papermc.io/repository/maven-public/") // Paper
+    maven("https://repo.convallyria.com/snapshots/")
+    maven("https://s01.oss.sonatype.org/content/repositories/snapshots/")
```
Comment (StealWonders) [inline build-logic/src/main/kotlin/buildlogic.java-common-conventions.gradle.kts:29]: This is used for?

**Review by StealWonders — CHANGES_REQUESTED** [review]

File: paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    override fun onDisable() {
+        // Plugin shutdown logic
+        webServer.stop()
+        if (webhookSender == null) return
+        webhookSender!!.stop()
+    }
+
+    private lateinit var loader: ConfigurationLoader<CommentedConfigurationNode>
```
[the remark this answers is not in the document]
Comment (SamB440) [reply paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt:84]: https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables

File: paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt (hunk 1 above)
Comment (SamB440) [reply paper/src/main/kotlin/org/empirewar/lemonadestand/paper/LemonadeStandPaper.kt:31]: I think you asked this before, iirc it's just weird kotlin, I think private set generates a setter, which we don't even need

File: build-logic/src/main/kotlin/buildlogic.java-common-conventions.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        indentWithSpaces(4)
+        trimTrailingWhitespace()
+    }
+}
+
+repositories {
+    // Use Maven Central for resolving dependencies.
+    mavenCentral()
+    maven("https://repo.papermc.io/repository/maven-public/") // Paper
+    maven("https://repo.convallyria.com/snapshots/")
```
[the remark this answers is not in the document]
Comment (SamB440) [reply build-logic/src/main/kotlin/buildlogic.java-common-conventions.gradle.kts:28]: See LemonSqueezer response

**Review by StealWonders — APPROVED** [review]

**Review by StealWonders — APPROVED** [review]

**Pull request merged**